ModularTegustation/tegu_items/workshop/_templates.dm

```diff
@@ -76,3 +81,40 @@
 		special_count += subject
 	else
 		special_count = subject
+
+/obj/item/ego_weapon/template/examine(mob/user)
+	. = ..()
+
+	if(level_xp==600)
+		. += "This weapon is fully upgraded!"
+		return
+
+
+	if(weapon_xp<level_xp)
+		. += "Weapon XP : [weapon_xp]/[level_xp]."
+	else
+		. += "Weapon XP : FULL! Bring to a workshop to reforge!"
+
+//Upgrade shit
+/obj/item/ego_weapon/template/attackby(obj/item/I, mob/living/user, params)
+	..()
+	if(istype(I, /obj/item/forginghammer))
+		if(!(locate(/obj/structure/table/anvil) in loc))
+			to_chat(user, span_warning("You need this to be on an anvil to work it."))
+			return
+
+		if(level_xp == 600)
+			to_chat(user, span_warning("This weapon cannot be upgraded."))
+			return
+
+		if(weapon_xp<level_xp)
```

Comment:
There's a lot of things like this, if ya could please give the code some space to breathe
```suggestion
		if(weapon_xp < level_xp)
```